### ISSUE TITLE
GitLab Support

### ISSUE THREAD

**Issue by tiokim** [Samsung/LPVS#93]
It would be great if LPVS supports GitLab as well  as Github.

**Comment by o-kopysov** [comment]
I will investigate GitLab webhooks and prepare a roadmap for implementation.

**Comment by o-konoval** [comment]
Official documentation from GitLab of webhooks:
https://docs.gitlab.com/ee/user/project/integrations/webhooks.html

**Comment by tiokim** [comment]
We may use https://github.com/gitlab4j/gitlab4j-api for this feature.

**Comment by o-kopysov** [comment]
> We may use https://github.com/gitlab4j/gitlab4j-api for this feature.

Yes, we already used this library in the [parser](https://github.sec.samsung.net/SOSHUB/Semantic-PoC/blob/repositories-parser/pom.xml), so we have the experience to use it.

**Comment by SoumyadipYT-OSS** [comment]
I am interested in Tizen .NET development

**Comment by yashpandey06** [comment]
@o-kopysov ...just for an overview are we trying to migrate the project to gitlab that's what the issue agenda is here ? ..correct me if i am wrong .

**Comment by SoumyadipYT-OSS** [comment]
Somebody found the issue or it will start from scratch?

**Comment by o-kopysov** [comment]
> Somebody found the issue or it will start from scratch?

No one has addressed this problem yet.

**Comment by SoumyadipYT-OSS** [comment]
@o-kopysov if you exaplain about this problem it would be great for me to understand the topic

**Comment by o-kopysov** [comment]
> [@o-kopysov](https://github.com/o-kopysov) if you exaplain about this problem it would be great for me to understand the topic

Currently, LPVS can provide scan result to user as a comment to GitHub pull request. You can refer to the source code in the file https://github.com/Samsung/LPVS/blob/main/src/main/java/com/lpvs/service/LPVSGitHubService.java 
Other VCSs are not supported right now.
So, our initial goal was to extend the list of supported VCS by implementing GitLab service class for doing similar functionality as currently implemented for GitHub. Because we currently have no human resources for this task, it was postponed.

**Comment by MoonkiHong** [comment]
@SoumyadipYT-OSS It would be great if you can contribute the above with our guidance, since we are out of resources as of now for delivering the feature above only with our team capacity.
